components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApisApiServiceImpl.java

```diff
@@ -2072,6 +2072,9 @@ private LifecycleStateDTO getLifecycleState(String apiId, String organization) {
                 RestApiUtil.handleResourceNotFoundError(RestApiConstants.RESOURCE_API, apiId, e, log);
             } else if (isAuthorizationFailure(e)) {
                 RestApiUtil.handleAuthorizationFailure("Authorization failure while deleting API : " + apiId, e, log);
+            } else if (e.getErrorHandler() != null && e.getErrorHandler() == ExceptionCodes.INVALID_API_ID) {
```

Comment:
This means we are already getting a correct error handler here? So just `} else { throw e }` should work I guess

Reply:
Fixed via https://github.com/wso2/carbon-apimgt/pull/11831/commits/83d71df99e5e6ccd90e74cd0e75c82ea8057a3f7